proposals/NNNN-reinstate-self-for-accessing-instance-properties-and-functions.md

```diff
@@ -0,0 +1,99 @@
+# Re-instate mandatory self for accessing instance properties and functions  
+
+* Proposal: [SE-NNNN](https://github.com/apple/swift-evolution/blob/master/proposals/NNNN-mandatory-self-for-accessing-instance-properties-and-functions.md)
+* Author(s): [David Hart](https://github.com/hartbit)
+* Status: **Review**
+* Review manager: TBD
+
+## Introduction
+
+Swift used to force using `self.` when accessing instance properties and functions, but since dropped this requirement (only keeping it for closure captures). It now feels inconsistent, and we have lost the automatic documentation of instance properties vs local variables and instance functions vs local functions or closures. This proposal offers to re-instate the original behaviour.
```

Comment:
You misunderstand me: when I said that it was enabled early in the Swift betas, I was talking about the betas for Xcode 6.3. It was then reverted in beta 3. The sentence I quoted is from the Xcode 6.3 beta 3 Release Notes which you can find here:

http://mazur.me/SwiftInFlux/docs/6.3-beta3.pdf > Page 6 > Swift Language